content/sensu-go/5.14/reference/agent.md

```diff
@@ -829,7 +829,7 @@ allow-list: /etc/sensu/check-allow-list.yaml{{< /highlight >}}
 
 | labels     |      |
 -------------|------
-description  | Custom attributes to include with event data, which can be accessed using [filters][9] and [tokens][27].<br><br>In contrast to annotations, you can use labels to create meaningful collections that can be selected with [API filtering][api-filter] and [sensuctl filtering][sensuctl-filter]. Overusing labels can impact Sensu's internal performance, so we recommend moving complex, non-identifying metadata to annotations.
+description  | Custom attributes to include with event data, which can be accessed using [filters][9] and [tokens][27].<br><br>In contrast to annotations, you can use labels to create meaningful collections that can be selected with [API filtering][api-filter] and [sensuctl filtering][sensuctl-filter].<br><br>Limit labels to metadata you need to use for response filtering. For complex, non-identifying metadata that you will *not* need to use in response filtering, use annotations rather than labels.
```

Comment:
In addition to response filtering in HTTP API requests and sensuctl commands, labels can be used to filter items in the web UI. Maybe that's also worth calling out here.

Something about `meaningful collections that can be selected` doesn't read well to me, but I don't have a better suggestion.

Reply:
Yeah, I don't like it either. Let me think about it...I will come up with something clearer.